lua/ttt2/extensions/table.lua

```diff
@@ -58,3 +58,26 @@ function table.SetWithPath(dataTable, path, value)
 	-- Set the value on the table (the last table on the path)
 	currentDataTable[path[#path]] = value
 end
+
+---
+-- Removes all empty table entries, making the table sequential again. 
+-- Use this function to more efficiently removing multiple indices from a sequential table by combining it
+-- with a function setting all entries to be removed to nil. Do NOT use table.Remove() or table.RemoveByValue()
+--
+-- @param dataTable table the table to traverse and set the value in.
+-- @param tableSize number the number of entries in dataTable
+function table.RemoveEmptyEntries(dataTable, tableSize)
```

Comment:
Why do I need to define `tableSize` on my own?

Reply:
Because ```#dataTable``` won't return the correct size (the table is no longer sequential due to the nil values) and ```table.Count(dataTable)``` is not efficient.